### PR TITLE
OCPBUGS-32426: [release-4.15] Improves service iptables efficiency on start up

### PR DESCRIPTION
**Pull request by trozet**
Only conflicts were in 2nd commit: "Fixes inefficient service parsing". I had to add a downstream carry commit in order to fix ordering for MCS rules and ensure they were at the top of the chain for unit tests.

### CODE REVIEW

**Comment by trozet** [comment]
/retitle OCPBUGS-32426: [release-4.15] Improves service iptables efficiency on start up

File: go-controller/pkg/node/gateway_localnet_linux_test.go [hunk 1]
```diff
@@ -627,9 +627,6 @@ var _ = Describe("Node Operations", func() {
 				fakeOvnNode.fakeExec.AddFakeCmd(&ovntest.ExpectedCmd{
 					Cmd: "ovs-ofctl show ",
 				})
-				fakeOvnNode.fakeExec.AddFakeCmd(&ovntest.ExpectedCmd{
-					Cmd: "ovs-ofctl show ",
-				})
 				service := *newService("service1", "namespace1", "10.129.0.2",
```
Comment (ricky-rav) [inline go-controller/pkg/node/gateway_localnet_linux_test.go:645]: Ah we used to run it twice... ok

**Review by ricky-rav — APPROVED** [review]
LGTM
Thanks, Tim!

**Pull request merged**